RFC.md

```diff
@@ -244,43 +244,60 @@ This means that  “foo => [bar, foo]” should become “foo => “ciphertext
 
-The client can verify whether it is a new file or an existing one by downloading the metadata and checking if the “files” array contain the referenced file.
-
-#### Uploading new files
-In case a new file is uploaded the client has to do the following steps:
-
-1. Generate a new 128-bit encryption key for the file and encrypt it using AES/GCM/NoPadding.
-2. Create new random identifier by generating a random UUID and removing the dash (`-`). The identifier must follow `/^[0-9a-fA-F]{32}$/`
-3. Upload the encrypted file via WebDAV using the identifier as file ID
-4. Add new file to the files array in the metadata file
-5. Update and lock the encrypted folder as described in “Update metadata file”. The latest metadataKey should be used to encrypt the metadata.
-
-#### Updating existing files
-In case an existing file is updated the client has to do the following steps:
-1. Generate a new 128-bit encryption key for the file and encrypt it using AES/GCM/NoPadding.
-2. Lock the encrypted folder
-3. Use the existing random identifier for the file and upload the encrypted file via WebDAV using the existing random identifier as file ID
-4. Update the file in the files array of the metadata
-5. Update and unlock the metadata file. The latest metadataKey should be used to encrypt the metadata.
-
-#### Accessing encrypted files 
+#### Update metadata file
+To keep the metadata and the file in sync locking is required.
+The client needs to lock the encrypted folder.
+If the lock operation succeeded the server responds with HTTP status code 200 together with a token in the response body.
+In case the client lost connection after locking the folder, it can restart the operation later with another "lock" request.
+In this case the client should send the token with the new lock call.
+This enables the server to decide if the client is allowed to retry the upload.
+
+After locking was successful, the client will upload the encrypted file and afterwards the metadata file.
+After both files are uploaded successfully the client will finish the operation by sending an unlock request.
+
+The `<folder-id>` denotes the ID of the end-to-end encrypted folder given by the WebDAV API.
+
+To lock the folder a POST request to `/ocs/v2.php/apps/end_to_end_encryption/api/v1/lock/<folder-id>` has to be sent.
+To add an existing lock token it can be sent as `token` parameter.
+
+To update the metadata a PUT request to `/ocs/v2.php/apps/end_to_end_encryption/api/v1/meta-data/<folder-id>` has to be sent.
+The request requires following two parameters:
+* The `token` contains the current lock token
+* The `metadata` contains the encrypted metadata JSON.
+
+To unlock the folder a DELETE request to `/ocs/v2.php/apps/end_to_end_encryption/api/v1/lock/<folder-id>` has to be sent.
+The previously received lock token has to be sent as `token` parameter.
+
+#### Modifying content of end-to-end encrypted folders
```

Comment:
I am not crazy about this - having separate sections for modifying a file, adding one, deleting one etc is easier, even if they are very similar...

Also, to get the conflict this causes with Roelands' other change resolved, it is probably best to revert this and make changes to the new text. This IS more complete, eg 
> Create new random identifier by generating a random UUID and removing the dash (`-`). The identifier must follow `/^[0-9a-fA-F]{32}$/`
That is good to have!

Reply:
It was scattered across multiple subsections before. It took me some time to figure out that most of the actions like adding files and changing files have many steps in common. That is the reason why I wrote it as nested enumeration. I think extra subsections would break this concise (and in my opinion easier/faster to understand) structure.